worker/lease/config.go

```diff
@@ -67,6 +67,10 @@ type ManagerConfig struct {
 	// case that the worker times out waiting to shut down.
 	LogDir string
 
+	// CancelDBOps is a cancel function that when called will cancel any
+	// context-bound database operations currently in progress.
+	CancelDBOps func()
```

Comment:
I guess the idea is that instead of any individual request getting cancelled, you instead cancel everything with a single request.
Hm...
That is absolutely how we've been using it. I'm just wondering if we wouldn't like ways to have individual threads get cancelled. (eg, the reason to proliferate `context.Context` everywhere is because 1 specific request can timeout because it is taking too long, without saying you need to reset your connection to the database.)

I think this is a net win, but I do think we should consider our timeout strategies and see if we want threaded `context.Context` instead. Especially nice with threading context is getting into opentracing.